vmm/src/config.rs

```diff
@@ -0,0 +1,252 @@
+// Copyright © 2019 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use linux_loader::cmdline::Cmdline;
+use net_util::MacAddr;
+use std::net::Ipv4Addr;
+use std::path::Path;
+use std::result;
+use vm_memory::{GuestAddress, GuestUsize};
+
+const DEFAULT_VCPUS: u8 = 1;
+const DEFAULT_MEMORY: GuestUsize = 512;
+pub const DEFAULT_RNG_SOURCE: &str = "/dev/urandom";
+const CMDLINE_OFFSET: GuestAddress = GuestAddress(0x20000);
+
+/// Errors associated with VM configuration parameters.
+#[derive(Debug)]
+pub enum Error {
+    /// Failed parsing cpus parameters.
+    ParseCpusParams,
+    /// Failed parsing memory parameters.
+    ParseMemoryParams,
+    /// Failed parsing kernel parameters.
+    ParseKernelParams,
+    /// Failed parsing kernel command line parameters.
+    ParseCmdlineParams,
+    /// Failed parsing disk parameters.
+    ParseDiskParams,
+    /// Failed parsing random number generator parameters.
+    ParseRngParams,
+    /// Failed parsing network parameters.
+    ParseNetParams,
+}
+pub type Result<T> = result::Result<T, Error>;
+
+pub struct VmParams<'a> {
+    pub cpus: Option<&'a str>,
```

Comment:
If you moved the default to the arg as a I suggest above you can drop the Option for this one and Memory and as a bonus the default will appear in the help text!